app/services/request_id_validator.rb

```diff
@@ -0,0 +1,37 @@
+class RequestIdValidator
+  include ActiveModel::Model
+  include ActionView::Helpers::TranslationHelper
+
+  validate :validate_request_id
+
+  def initialize(request_id)
+    @request_id = request_id
+  end
+
+  def submit
+    success = valid?
+
+    FormResponse.new(
+      success: success,
+      errors: errors,
+      extra: extra_attributes
+    )
+  end
+
+  private
+
+  attr_reader :request_id
+
+  def validate_request_id
+    return if request_id.blank?
+    return if ServiceProviderRequest.where(uuid: request_id).exists?
+
+    errors.add(:request_id, :invalid)
+  end
+
+  def extra_attributes
+    {
+      request_id_present: request_id.present?
```

Comment:
How about capturing the actual request_id to help us troubleshoot? I think we should also capture it when it first gets set. That way, we can compare and see if it's being modified on the password screen.